service/frontend/workflow_handler.go

```diff
@@ -3168,44 +3137,38 @@ func (wh *WorkflowHandler) UpdateSchedule(ctx context.Context, request *workflow
 		return nil, err
 	}
 
-	input := &schedspb.FullUpdateRequest{
+	getStartInpuPayloads := func(startScheduleArgs *schedspb.StartScheduleArgs) (inputPayloads *commonpb.Payloads, err error) {
+		if startScheduleArgs.Schedule != nil {
+			startScheduleArgs.Schedule.Spec = request.Schedule.GetSpec()
+			startScheduleArgs.Schedule.Action = request.Schedule.GetAction()
+			startScheduleArgs.Schedule.Policies = request.Schedule.GetPolicies()
+			startScheduleArgs.Schedule.State = request.Schedule.GetState()
+		}
+		// don't touch Info
+		if startScheduleArgs.Info != nil {
+			startScheduleArgs.Info.UpdateTime = timestamp.TimePtr(time.Now().UTC())
+		}
+
+		inputPayloads, err = sdk.PreferProtoDataConverter.ToPayloads(startScheduleArgs)
```

Comment:
do you think we could move the ToPayloads calls (for start args + signal) into signalWithStartScheduleWorkflow? they're repeated in patch...